tests/containers/skopeo.pm

```diff
@@ -90,10 +93,57 @@ sub run {
     record_info('Compare images', 'Both extracted copies must be identical.');
     assert_script_run("diff -urN $dir1 $dir2", fail_message => 'Copied images are not identical.');
 
-    # Add cleanup routine.
+    ######### Spin-up an instance of the latest Registry
+    assert_script_run("podman run --rm -d -p 5050:5000 --name skopeo-registry registry.suse.com/suse/registry:latest",
+        fail_message => "Failed to start local registry container");
+
+    ######### Pull the image into a our local repository
+    # skipping tls verification as by default most local registries don't have certificates
+    record_info('Copy Image', 'Copy image from remote repository into the local repository.');
+    validate_script_output("skopeo copy --dest-tls-verify=0 docker://$remote_image docker://$local_image",
+        sub { m/Writing manifest to image destination/ },
+        fail_message => 'Failed to copy image to local repository.');
+
+    ######### Inspect the local image repository
+    # skipping tls verification as by default most local registries don't have certificates
+    record_info('Inspect Image', 'Inspect an image from the local repository.');
+    assert_script_run("skopeo inspect --tls-verify=0 docker://$local_image",
+        fail_message => 'Failed to inspect local image.');
+
+    ######### Compare remote image to local image
+    # using JQ to omit any repository-specific fields which are expected to be different
+    record_info('Verify Remote Image', 'Inspect remote image and save results.');
+    assert_script_run("skopeo inspect --tls-verify=0 docker://$remote_image | jq .Layers[] >> $workdir/inspect_remote.json",
+        fail_message => 'Failed to inspect remote image.');
+
+    record_info('Verify Local Image', 'Inspect local image and save results.');
+    assert_script_run("skopeo inspect --tls-verify=0 docker://$local_image | jq .Layers[] >> $workdir/inspect_local.json",
+        fail_message => 'Failed to inspect local image.');
+
+    record_info('Compare local and remote images', 'Compare local and remote images.');
+    assert_script_run("diff $workdir/inspect_remote.json $workdir/inspect_local.json",
+        fail_message => 'Images are not identical!');
+}
+
+# Add cleanup routine.
```

Comment:
The name of the routine makes this comment a little bit redundant, imho.

Reply:
Very much so :)